### PR TITLE
Add hook for ASG that supports all scaling processes

### PR DESCRIPTION
**Pull request by pinkieee**
This sceptre hook can suspend/resume any autoscaling processes as they are defined in the AWS documentation:

http://docs.aws.amazon.com/autoscaling/latest/userguide/as-suspend-resume-processes.html

### CODE REVIEW

**Comment by jamesroutley** [comment]
Nice! Thanks for contributing - I'll take a look through it tomorrow.

File: sceptre/hooks/asg_scaling_processes.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    useful as scheduled actions must be suspended when updating stacks with
+    autoscaling groups.
+    This command supports all autoscaling scaling processes.
+    Example configuration:
+    hooks:
+      before_update:
+        - !asg_scaling_processes suspend::ScheduledActions
+    Full documentation on the suspend and resume processes
+    http://docs.aws.amazon.com/autoscaling/latest/userguide/as-suspend-resume-processes.html
+    """
```
Comment (theseanything) [inline sceptre/hooks/asg_scaling_processes.py]: Can we add a rst format code block around the example?

File: sceptre/hooks/asg_scaling_processes.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    def __init__(self, *args, **kwargs):
+        super(ASGScalingProcesses, self).__init__(*args, **kwargs)
+
+    def run(self):
+        """
+        Either suspends or resumes any scaling processes on all autoscaling
+        groups with in the current stack.
+        """
```
Comment (theseanything) [inline sceptre/hooks/asg_scaling_processes.py]: Also add a rst docstring raises bit at the end... `:raises: InvalidHookArgumentSyntaxError`

File: sceptre/hooks/asg_scaling_processes.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+from sceptre.hooks import Hook
+from sceptre.exceptions import InvalidHookArgumentTypeError
+from sceptre.exceptions import InvalidHookArgumentSyntaxError
+from sceptre.exceptions import InvalidHookArgumentValueError
+
+
+class ASGScalingProcesses(Hook):
+    """
+    A command to resume or suspend autoscaling group scaling processes. This is
```
Comment (theseanything) [inline sceptre/hooks/asg_scaling_processes.py]: Can remove the `A command to...` just have `Resumes or suspends....`

File: sceptre/hooks/asg_scaling_processes.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    useful as scheduled actions must be suspended when updating stacks with
+    autoscaling groups.
+    This command supports all autoscaling scaling processes.
+    Example configuration:
```
Comment (theseanything) [inline sceptre/hooks/asg_scaling_processes.py]: Maybe move the example to the sceptre docs  - but include note about argument syntax `!asg_scaling_processes <suspend|resume>::<process-name>`?

File: sceptre/hooks/asg_scaling_processes.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+        if not isinstance(self.argument, basestring):
+            raise InvalidHookArgumentTypeError(
+                'The argument "{0}" is the wrong type - asg_scaling_processes '
+                'hooks require arguments of type string.'.format(self.argument)
+            )
+        if "::" not in str(self.argument):
+            raise InvalidHookArgumentSyntaxError(
+                'Wrong syntax for the argument "{0}" - asg_scaling_processes '
+                'hooks use the format: "suspend::Terminate".'.format(self.argument)
```
Comment (theseanything) [inline sceptre/hooks/asg_scaling_processes.py]: maybe generalise argument format like above in docstring?

File: tests/test_hooks/test_asg_scaling_processes.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+            _find_autoscaling_groups()
+
+        assert response == []
+
+    @patch(
+        "sceptre.hooks.asg_scaling_processes"
+        ".ASGScalingProcesses._find_autoscaling_groups"
+    )
+    def test_run_with_resume_argument(self, mock_find_autoscaling_groups):
+        self.mock_asg_scaling_processes.argument = u"resume::ScheduledActions"
```
Comment (theseanything) [inline tests/test_hooks/test_asg_scaling_processes.py]: Is the unicode string for python3?

**Comment by theseanything** [comment]
@jamesroutley just want to add this would also carry out functionality of the existing `asg_scheduled_actions` hook. So we could probably remove it, but would need to think whether to wait until a major release....

File: tests/test_hooks/test_asg_scaling_processes.py (hunk 6 above)
Comment (theseanything) [inline tests/test_hooks/test_asg_scaling_processes.py]: @jamesroutley do you know?

**Comment by theseanything** [comment]
Everything is great, just need to work out why there are explicit unicode strings (`u'string'`) in some of the tests.

**Comment by theseanything** [comment]
need separate issue/pr for unicode string - not sorted in this pr as out of scope and is present in other testing code.

**Pull request merged**